### PR TITLE
netmaker: init at 0.17.1

### LINKED ISSUE #207654: NetClient a NetMaker client.
**NetMaker and NetClient**
Netmaker is a networking platform built on the groundbreaking WireGuard protocol. Netmaker unifies your hybrid/multi-cloud, edge, IoT, and Kubernetes environments.

**Metadata**

* homepage URL: https://www.netmaker.org/
* source URL: https://github.com/gravitl/netmaker/releases
* license: mit, bsd, gpl2+ , SSPL
* platforms: unix, linux, darwin, 

Would like to request a nix package for NetClient, but both NetMaker and NetClient would also be nice.

### PR DESCRIPTION
**Pull request by urandom2**
###### Description of changes

Fixes #207654

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/netmaker/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    xorg.libXinerama
+    xorg.libXrandr
+  ];
+
+  nativeBuildInputs = [pkg-config];
+
+  meta = with lib; {
+    description = "WireGuard automation from homelab to enterprise";
+    homepage = "https://netmaker.io";
+    changelog = "https://github.com/${src.owner}/${src.repo}/-/releases/v${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/netmaker/default.nix:35]:
```suggestion
    changelog = "https://github.com/gravitl/netmaker/-/releases/v${version}";
```

File: pkgs/applications/networking/netmaker/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  buildInputs = [
+    libglvnd
+    xorg.libX11
+    xorg.libXcursor
+    xorg.libXi
+    xorg.libXinerama
+    xorg.libXrandr
+  ];
+
+  nativeBuildInputs = [pkg-config];
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/netmaker/default.nix:30]:
```suggestion
  nativeBuildInputs = [ pkg-config ];

  buildInputs = [
    libglvnd
    xorg.libX11
    xorg.libXcursor
    xorg.libXi
    xorg.libXinerama
    xorg.libXrandr
  ];
```

**Comment by urandom2** [comment]
we certainly can break them apart, but much like coreutils, this seems like something that will be easier to build once and get both for free. let me know if two derivations is a hard ask on this one?

**Comment by SuperSandro2000** [comment]
We should probably add an option to only build the server without x and then we can add that to top-level.

**Pull request merged**